Library/Homebrew/tap.rb

```diff
@@ -877,6 +880,8 @@ def should_report_analytics?
 
   sig { params(other: T.nilable(T.any(String, Tap))).returns(T::Boolean) }
   def ==(other)
+    return false if other.nil?
+
     other = Tap.fetch(other) if other.is_a?(String)
     self.class == other.class && name == other.name
   end
```

Comment:
Is this a typechecker shortcoming thing? `other.is_a?` and the class check should cover it.

Reply:
That only checks if it is a `String`, but it can still be `nil`.

Reply:
I wonder where we even use this check, i.e. compare a `Tap` to a `String`, rather than just comparing `Tap#name` explicitly.

Reply:
> That only checks if it is a `String`, but it can still be `nil`.

Yeah, at which point it'll do `self.class == nil.class` which is always false

Reply:
I guess that part is what Sorbet doesn't know.

Reply:
Could just do `T.must(other).name` then.

Reply:
`other.is_a?(self.class` seems to work.

Reply:
Does subtly change the behaviour though so CoreTap.instance == Tap.new("Homebrew", "core") is true now, albeit minor.

Reply:
Yes, but we should never have a `Tap.new("Homebrew", "core")` in the first place.

The only reason why `Tap::new` isn't private is because it's used in tests.

Reply:
I guess they can all use `fetch` instead, I don't think we have a unit test for `Tap::new` specifically anyways.